internal/authz/authz_test.go

```diff
@@ -58,3 +73,181 @@ func TestAllRolesExistInFGAModel(t *testing.T) {
+
+	// ensure assignments to project returns no assignments
+	assignments, err = c.AssignmentsToProject(userctx, prj)
+	assert.NoError(t, err, "failed to get assignments to project")
+	assert.Len(t, assignments, 0, "expected 0 assignments to project")
+}
+
+func TestVerifyMultipleProjects(t *testing.T) {
+	t.Parallel()
+
+	c, stopFunc := newOpenFGAServerAndClient(t)
+	defer stopFunc()
+	assert.NotNil(t, c)
+
+	ctx := context.Background()
+
+	assert.NoError(t, c.MigrateUp(ctx), "failed to migrate up")
+
+	// this is required to auto-detect the generated model and store
+	assert.NoError(t, c.PrepareForRun(ctx), "failed to prepare for run")
+
+	// create a project
+	prj1 := uuid.New()
+	assert.NoError(t, c.Write(ctx, "user-1", authz.AuthzRoleAdmin, prj1), "failed to write project")
+
+	userctx := auth.WithUserSubjectContext(ctx, "user-1")
+
+	// verify the project
+	assert.NoError(t, c.Check(userctx, "get", prj1), "failed to check project")
+
+	// create another project
+	prj2 := uuid.New()
+	assert.NoError(t, c.Write(ctx, "user-1", authz.AuthzRoleViewer, prj2), "failed to write project")
+
+	// verify the project
+	assert.NoError(t, c.Check(userctx, "get", prj2), "failed to check project")
+
+	// create an unrelated project
+	prj3 := uuid.New()
+	assert.NoError(t, c.Write(ctx, "user-2", authz.AuthzRoleAdmin, prj3), "failed to write project")
+
+	// verify the project
+	assert.NoError(t, c.Check(auth.WithUserSubjectContext(ctx, "user-2"), "get", prj3), "failed to check project")
+
+	// verify user-1 cannot operate on project 3
+	assert.Error(t, c.Check(userctx, "get", prj3), "expected user-1 to not be able to operate on project 3")
+
+	// ensure projects for user returns the projects
+	projects, err := c.ProjectsForUser(userctx, "user-1")
+	assert.NoError(t, err, "failed to get projects for user")
+	assert.Len(t, projects, 2, "expected 2 projects for user")
+	assert.Contains(t, projects, prj1, "expected project to be returned")
+	assert.Contains(t, projects, prj2, "expected project to be returned")
+
+	// ensure assignments to project returns the user
+	assignments, err := c.AssignmentsToProject(userctx, prj1)
+	assert.NoError(t, err, "failed to get assignments to project")
+	assert.Len(t, assignments, 1, "expected 1 assignment to project")
+	assert.Equal(t, "user-1", assignments[0].Subject, "expected user to be assigned to project")
+
+	// ensure assignments to project returns the user
+	assignments, err = c.AssignmentsToProject(userctx, prj2)
+	assert.NoError(t, err, "failed to get assignments to project")
+	assert.Len(t, assignments, 1, "expected 1 assignment to project")
+	assert.Equal(t, "user-1", assignments[0].Subject, "expected user to be assigned to project")
+
+	// Delete the user (which also deletes the projects)
+	assert.NoError(t, c.DeleteUser(ctx, "user-1"), "failed to delete user")
+
+	// verify the projects are gone
+	assert.Error(t, c.Check(userctx, authz.AuthzRoleAdmin.String(), prj1), "expected project to be gone")
```

Comment:
Should this be "get" or "update" here as well?

Reply:
oops! missesd a spot.